app/src/main/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailViewModel.kt

```diff
@@ -32,19 +45,67 @@ class DetailViewModel @Inject constructor(
     override val coinDetailUiModel: StateFlow<CoinDetailUiModel?>
         get() = _coinDetailUiModel
 
+    private val _coinPrices = MutableStateFlow<List<CoinPrice>>(emptyList())
+    override val coinPrices: StateFlow<List<CoinPrice>>
+        get() = _coinPrices
+
     override fun getCoinId(coinId: String) {
         getCoinDetail(coinId = coinId)
+        getCoinPrices(coinId = coinId)
     }
 
     private fun getCoinDetail(coinId: String) = execute {
         showLoading()
         getCoinDetailUseCase.execute(coinId = coinId)
             .catch { e ->
+                hideLoading()
                 _error.emit(e)
             }
             .collect { coinDetail ->
+                hideLoading()
                 _coinDetailUiModel.emit(coinDetail.toUiModel())
             }
         hideLoading()
     }
+
+    @Suppress("MagicNumber")
+    fun getCoinPrices(coinId: String, timeIntervals: TimeIntervals = TimeIntervals.ONE_DAY) =
+        execute {
+            val calendar = Calendar.getInstance()
+            val fromTimestamp = when (timeIntervals) {
+                TimeIntervals.ONE_DAY -> calendar.apply {
+                    add(Calendar.DAY_OF_YEAR, -1)
+                }
+                TimeIntervals.ONE_WEEK -> calendar.apply {
+                    add(Calendar.DAY_OF_YEAR, -7)
+                }
+                TimeIntervals.ONE_MONTH -> calendar.apply {
+                    add(Calendar.MONTH, -1)
+                }
+                TimeIntervals.ONE_YEAR -> calendar.apply {
+                    add(Calendar.YEAR, -1)
+                }
+                TimeIntervals.FIVE_YEAR -> calendar.apply {
+                    add(Calendar.YEAR, -5)
+                }
+            }
```

Comment:
we can optimize this a bit
```suggestion
            val fromTimestamp = Calendar.getInstance().apply {
                when (timeIntervals) {
                    TimeIntervals.ONE_DAY -> add(Calendar.DAY_OF_YEAR, -1)
                    TimeIntervals.ONE_WEEK -> add(Calendar.DAY_OF_YEAR, -7)
                    TimeIntervals.ONE_MONTH -> add(Calendar.MONTH, -1)
                    TimeIntervals.ONE_YEAR -> add(Calendar.YEAR, -1)
                    TimeIntervals.FIVE_YEAR -> add(Calendar.YEAR, -5)
                }
            }
```